WordPress/gutenberg
# [RNMobile] Search Block: Add block toolbar options

## Description
This PR builds on #28983 by adding the block toolbar options. All options work, but since some Block UI elements are still yet to be implemented, the UI will not reflect those changes (even though the updates to the block attributes still work). To get around this I have added a temporary alert when the following options change to notify the tester that the change has completed successfully:
- Button position change
- Icon only button toggled
For reasons I'm unsure of, the alert message will pop up and dismiss immediately on iOS. If it's difficult to catch the message, changes can also be verified by switching to the HTML view and verifying the option change there. 

## How has this been tested?



Tested in the demo and WordPress apps for both iOS and Android platforms. 

Add a search block to the editor by selecting it from the block inserter, then continue with the following steps. Each section is broken up by the toolbar option under test for clarity. The search block currently only has a single UI element: the search label. This is how it looks when first added:

<img src="https://user-images.githubusercontent.com/5810477/107843530-e4bdcd00-6d99-11eb-99a1-b4f85666d2cd.png" width="300"/>

### Test toggle label

![Screen Shot 2021-02-13 at 1 23 21 AM](https://user-images.githubusercontent.com/5810477/107843555-2484b480-6d9a-11eb-9add-1f0c514b3287.png)

1. When the search block is first added, the "Toggle search label" option will already be toggled "on". 
2. Tap to toggle the search label option off. Verify the "Search" string in the block disappears. 
3. Tap to toggle the search label option on. Verify the "Search" string in the block reappears.


### Test button position

![Screen Shot 2021-02-13 at 1 24 29 AM](https://user-images.githubusercontent.com/5810477/107843575-5138cc00-6d9a-11eb-8d52-df97d82d6132.png)

1. Tap the **button position** option and select **No button**. Verify the alert matches the selection _and_ that the **icon only** toolbar option has been removed. 
2. Tap the **button position** option and select **Button inside**. Verify the alert matches the selection _and_ that the **icon only** toolbar option has been restored.
3. Tap the **button position** option and select **Button outside**. Verify the alert matches the selection.


**Menu options**

<img src="https://user-images.githubusercontent.com/5810477/107843579-55fd8000-6d9a-11eb-8026-f3749ae3acfc.png" width="300"/>

### Test icon only
1. Tap the **icon only** option to toggle it on. Verify the alert matches the selection.
2. Tap the **icon only** option to toggle it off. Verify the alert matches the selection.

![Screen Shot 2021-02-13 at 1 24 54 AM](https://user-images.githubusercontent.com/5810477/107843584-5d248e00-6d9a-11eb-85c0-3262f2690e95.png)


## Screenshots 

iOS | Android
-- | --
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/5810477/107844011-ec7f7080-6d9d-11eb-8126-66f295989c32.gif)|![ezgif com-gif-maker](https://user-images.githubusercontent.com/5810477/107844014-ee493400-6d9d-11eb-8665-ad87bef3e571.gif)




## Types of changes




New feature - continues to build on the Search block feature which will span multiple PRs.

## Checklist:
- [ ] I've tested my changes with keyboard and screen readers (this will come in a later PR specifically for accessibility)
